envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/EnvoyLoadBalancingPrioritiesTest.kt

```diff
@@ -0,0 +1,189 @@
+package pl.allegro.tech.servicemesh.envoycontrol
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.RegisterExtension
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.isEitherFrom
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.isFrom
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.isOk
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.untilAsserted
+import pl.allegro.tech.servicemesh.envoycontrol.config.consul.ConsulClusterSetup
+import pl.allegro.tech.servicemesh.envoycontrol.config.consul.ConsulMultiClusterExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.envoy.EnvoyExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.envoycontrol.EnvoyControlClusteredExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.service.EchoServiceExtension
+import java.time.Duration
+
+class EnvoyLoadBalancingPrioritiesTest {
+    companion object {
+        private const val serviceName = "echo"
+        private val pollingInterval = Duration.ofSeconds(1)
+        private val stateSampleDuration = Duration.ofSeconds(1)
+        private val properties = mapOf(
+            "envoy-control.envoy.snapshot.stateSampleDuration" to stateSampleDuration,
+            "envoy-control.sync.enabled" to true,
+            "envoy-control.sync.polling-interval" to pollingInterval.seconds,
+            "envoy-control.envoy.snapshot.loadBalancing.priorities.enabled" to true,
+            "envoy-control.envoy.snapshot.loadBalancing.priorities.dcPriorityProperties.dc1" to 0,
+            "envoy-control.envoy.snapshot.loadBalancing.priorities.dcPriorityProperties.dc2" to 1,
+            "envoy-control.envoy.snapshot.loadBalancing.priorities.dcPriorityProperties.dc3" to 2
+        )
+
+        @JvmField
+        @RegisterExtension
+        val consulClusters = ConsulMultiClusterExtension()
+
+        @JvmField
+        @RegisterExtension
+        val envoyControl =
+            EnvoyControlClusteredExtension(consulClusters.serverFirst, { properties }, listOf(consulClusters))
+
+        @JvmField
+        @RegisterExtension
+        val envoyControl2 =
+            EnvoyControlClusteredExtension(consulClusters.serverSecond, { properties }, listOf(consulClusters))
+
+        @JvmField
+        @RegisterExtension
+        val envoyControl3 =
+            EnvoyControlClusteredExtension(consulClusters.serverThird, { properties }, listOf(consulClusters))
+
+        @JvmField
+        @RegisterExtension
+        val envoy = EnvoyExtension(envoyControl)
+
+        @JvmField
+        @RegisterExtension
```

Comment:
there is no need to create 4 instances per dc, you can configure [this property](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/overprovisioning#arch-overview-load-balancing-overprovisioning-factor)